scipy/stats/tests/test_hypotests.py

```diff
@@ -159,3 +160,79 @@ def test_callable_cdf(self):
         r1 = cramervonmises(x, distributions.beta.cdf, args)
         r2 = cramervonmises(x, "beta", args)
         assert_equal((r1.statistic, r1.pvalue), (r2.statistic, r2.pvalue))
+
+
+class TestCvm_2samp(object):
+    def test_invalid_input(self):
+        x = np.arange(10).reshape((2, 5))
+        y = np.arange(5)
+        assert_raises(ValueError, cramervonmises_2samp, x, y)
+        assert_raises(ValueError, cramervonmises_2samp, y, x)
+        assert_raises(ValueError, cramervonmises_2samp, [], y)
+        assert_raises(ValueError, cramervonmises_2samp, y, [1])
+        assert_raises(ValueError, cramervonmises_2samp, y, y, 'wrong_method')
```

Comment:
Consider converting these to:
`with assert_raises(ValueError, match="error message here"):
   cramervonmises_2samp(...)`
There are a lot of `ValueErrors` out there, so it's good to know the right one is being raised. It also helps me follow what the test is checking for. (I think the first two fail because `x` is 2d, the next because the first input is empty, the next because the second input is size 1, and the last for not being "exact" or "asymp". Are size 1 inputs not valid for this test, or does it turn into a special case that's not worth handling?)

Reply:
I just excluded input with a single element since under H0, x < y or x > y with prob. 0.5, so it does not provide any useful information and i thought it is easier to simply raise an error instead of adding further test cases.

Reply:
added a check of the error message